src/simplify/DouglasPeuckerSimplifier.cpp

```diff
@@ -105,12 +105,11 @@ DPTransformer::transformCoordinates(
 {
     ::geos::ignore_unused_variable_warning(parent);
 
-    const Coordinate::Vect* inputPts = coords->toVector();
-    assert(inputPts);
+    Coordinate::Vect inputPts;
+    coords->toVector(inputPts);
```

Comment:
also here, if possible

```cpp
 coords->toVector(std::move(inputPts));
```

Reply:
Sorry, I don't understand :(. Are you suggesting to change the signature of `CoordinateSequence::toVector(std::vector<Coordinate> & v)` ?

Reply:
Oh, sorry, my bad.
I meant to write

```cpp
auto inputPts = coords->toVector();
```